### PR TITLE
Feat: Add custom user agent for Monika

### LINKED ISSUE #1292: Create a custom user agent for Monika
While monitoring some sites that use Cloudflare/LiteSpeed, there may be false incidents regarding Cloudflare anti-DDOS protection, which could return 403.

References:
https://betterstack.com/docs/uptime/frequently-asked-questions/#what-user-agent-does-uptime-use
https://betterstack.com/community/questions/whitelisting-better-stack-ip-cloudflare/

### PR DESCRIPTION
**Pull request by dennypradipta**
# Monika Pull Request (PR)

## What feature/issue does this PR add

This PR adds custom user agent when probing websites, and will fixes #1292 

## How did you implement / how did you fix it

1. Add logic for handling user agents from config
2. Add default value for user agent
3. Add flag for user agent

## How to test

1. Create a simple fastify server:
```js
// Import the framework and instantiate it
import Fastify from "fastify";

const fastify = Fastify({
  logger: {
    serializers: {
      req(req) {
        console.log(req.headers["user-agent"]);

        return {
          userAgent: req.headers["user-agent"],
        };
      },
    },
  },
});

// Declare a route
fastify.get("/", async function handler(request, reply) {
  return { hello: "world" };
});

// Run the server!
try {
  await fastify.listen({ port: 3000 });
} catch (err) {
  fastify.log.error(err);
  process.exit(1);
}

```
2. Run the server
3. Run Monika pointing to the local server
4. Pay attention to the user agent

## Video
https://github.com/hyperjumptech/monika/assets/16670475/67920681-af24-47e2-8117-a6c67c7cf972

### CODE REVIEW

File: src/flag.ts [hunk 1]
```diff
@@ -100,6 +101,8 @@ export const monikaFlagsDefaultValue: MonikaFlags = {
   symonGetProbesIntervalMs: 60_000,
   symonReportInterval: DEFAULT_SYMON_REPORT_INTERVAL_MS,
   symonReportLimit: 100,
+  'user-agent':
+    'Monika Synthetic Monitoring Bot Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.169 Safari/537.36',
```
Comment (haricnugraha) [inline src/flag.ts:105]: I prefer a simpler user-agent with relevant content, such as `monika/{version}`. For instance, `monika/1.20.2`.

Comment (dennypradipta) [reply src/flag.ts]: Changed it to monika/1.20.2, according to the current npm version

**Comment by dennypradipta** [comment]
Still verifying whether if the Docker version uses the custom user agent.

File: src/flag.ts (hunk 1 above)
Comment (sapiderman) [reply src/flag.ts]: ooh never mind: recommended is full version:

https://developers.whatismybrowser.com/learn/browser-detection/user-agents/user-agent-best-practices

**Review by sapiderman — APPROVED** [review]
LGTM

**Review by ilmiawan — APPROVED** [review]

**Comment by haricnugraha** [comment]
I just remembered Monika has `getContext().userAgent`. The value is `@hyperjumptech/monika/1.20.2 (darwin-x64) node-20.11.0`.

**Comment by dennypradipta** [comment]
> I just remembered Monika has `getContext().userAgent`. The value is `@hyperjumptech/monika/1.20.2 (darwin-x64) node-20.11.0`.

Thanks for the tip! It works flawlessly

Docker
image

NPM
image

**Comment by dennypradipta** [comment]
image

Here is the example for using custom user agent

File: docs/src/pages/guides/cli-options.md [hunk 2]
```diff
@@ -347,6 +347,15 @@ Only usable for probes which does not have [chaining requests.](https://hyperjum
 monika --ttl-cache 5
 ```
 
+## User Agent
+
+The --user-agent flag in the Monika command allows you to set a custom user agent string for HTTP requests. This string is used to identify the browser, operating system, and device making the request. Setting a custom user agent can help simulate different browsing environments or bypass certain server restrictions. Defaults to "monika/{current Monika version}"
+
+```bash
+# Set TTL cache for HTTP to 5 minutes
```
Comment (haricnugraha) [inline docs/src/pages/guides/cli-options.md:355]: Please adjust the documentation.

**Review by haricnugraha — APPROVED** [review]

File: docs/src/pages/guides/cli-options.md (hunk 2 above)
Comment (dennypradipta) [reply docs/src/pages/guides/cli-options.md:355]: I got you

**Pull request merged**